### DISCUSSION TITLE
QDrawer persistency

### DISCUSSION THREAD

**Discussion by fmarzocca** [quasarframework/quasar#13850] [General - Components / Directives / etc]
I would like that my QDrawer closes when I change route, but it is always on view.

This is my html:

```
 <q-drawer v-model="drawerLeft" :width="200" elevated bordered class="bg-grey-3" :breakpoint="500" :persistent="isPersist">
                <q-scroll-area class="fit">
                    <template v-for="(menuItem, index) in menuList" :key="index">
                        <q-item clickable :active="menuItem.label" :to="menuItem.to" v-ripple>
                            <q-item-section avatar>
                                <q-icon :name="menuItem.icon" />
                            </q-item-section>
                            <q-item-section>
                                {{ menuItem.label }}
                            </q-item-section>
                        </q-item>
                        <q-separator :key="'sep' + index" v-if="menuItem.separator" />
                    </template>
                </q-scroll-area>
            </q-drawer>
```

and the variable `isPersist` is handled in `data`, as always false:

`isPersist: false,`

but the drawer is persistent.

**Comment by MilosPaunovic** [comment]
If it's always false, why not just remove it from both `data` and the component prop?

**Reply by fmarzocca** [reply]
> why not just remove it from both `data` and the component prop?

because that was the first attempt, and it didn't work.

**Reply by MilosPaunovic** [reply]
Are you replacing route or pushing a new one?

**Comment by fmarzocca** [comment]
I am using the `:to` selector in the menu:

```
 menuList: [
                {
                    icon: 'home',
                    label: 'Home',
                    to: '/',
                    separator: false
                },
                {
                    icon: 'inbox',
                    label: 'Devices',
                    to: '/devices',
                    separator: true
                }
     ]
```


```
<template v-for="(menuItem, index) in menuList" :key="index">
                        <q-item clickable :active="menuItem.label" :to="menuItem.to" v-ripple>
```

**Comment by fmarzocca** [comment]
It would be also nice if the drawer closes when I click on the page, without the need to click the button.

**Comment by MilosPaunovic** [comment]
[https://stackblitz.com/edit/quasarframework-webpack-udjdbs?file=src/layouts/MainLayout.vue](https://stackblitz.com/edit/quasarframework-webpack-udjdbs?file=src/layouts/MainLayout.vue)

I couldn't reproduce your issue. Please provide proper reproduction, either on StackBlitz or a GitHub repository for more support.

**Reply by fmarzocca** [reply]
You are testing in small device, so the drawer switches to "mobile" behavior. If you widen the window (by sliding it to the left), you will see that the drawer is persistent. Your stackblitz if perfectly reproducing the issue.

**Comment by MilosPaunovic** [comment]
Okay, solution to your issue would be to add ` behavior="mobile` and remove `:breakpoint="500"` & `:persistent="isPersist"` from your `QDrawer`.

**Reply by fmarzocca** [reply]
Understand. But this is not a solution, just a workaround to force the drawer acting as It was on a mobile. So, Quasar docs are wrong on this componenti.
